src/Data/Reader/EntityReader.php

```diff
@@ -184,14 +185,14 @@ private function setFilterProcessors(FilterProcessorInterface ...$filterProcesso
         $this->filterProcessors = array_merge($this->filterProcessors, $processors);
     }
 
-    private function buildQuery()
+    private function selectQuery(): SelectQuery|Select
```

Comment:
This method builds a select query but not selects a query.
If you don't like the `buildQuery` method name then you can set it like `buildSelectQuery`. 
For the consistency! :)